core/mutator/entry/mutation.go

```diff
@@ -56,11 +57,12 @@ func NewMutation(index []byte, directoryID, userID string) *Mutation {
 
 // SetPrevious sets the previous hash.
 // If copyPrevious is true, AuthorizedKeys and Commitment are also copied.
-func (m *Mutation) SetPrevious(oldValue []byte, copyPrevious bool) error {
+func (m *Mutation) SetPrevious(rev uint64, oldValue []byte, copyPrevious bool) error {
```

Comment:
The target of the revision here is potentially ambiguous. My context-free instinct would be that this revision should be set to the revision at which the value was set to oldValue. My understanding from the rest of the code in this CL is that rev is actually intended to be set to the current live/latest revision for the map. This could be a much higher revision than when the value was last modified.

On further reading of waitOnceForUserUpdate I think either revision would be fine. Leaving this comment as I think a little clarity on expectation here would be beneficial.

Reply:
The intent here is to give the client a clue about how long to wait before attempting to verify that a mutation was applied correctly.   This behavior works best when `rev` is set to the revision at which `oldValue` was fetched since the client will wait until the server's reported current revision > `rev`.